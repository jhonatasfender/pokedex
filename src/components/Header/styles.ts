import styled from "styled-components";

export const Wrapper = styled.header`
  padding: 1rem;

  img {
    width: 17rem;
  }
`;
