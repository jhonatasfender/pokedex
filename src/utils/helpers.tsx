import React from "react";
import { ThemeProvider } from "styled-components";

import { render, RenderResult } from "@testing-library/react";

import theme from "../style-global/theme";

export const renderWithTheme = (children: React.ReactNode): RenderResult =>
  render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
