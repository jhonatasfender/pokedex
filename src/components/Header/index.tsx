import * as S from "./styles";

export default function Header() {
  return (
    <S.Wrapper>
      <img src="/img/logo.png" alt="" />
    </S.Wrapper>
  );
}
