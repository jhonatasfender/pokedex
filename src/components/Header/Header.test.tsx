import { screen } from "@testing-library/react";

import { renderWithTheme } from "../../utils/helpers";
import Header from "./";

describe("Header", () => {
  it("should initialize with default parameters", () => {
    renderWithTheme(<Header />);

    const header = screen.getByRole("banner");

    expect(header).toMatchSnapshot();
  });
});
