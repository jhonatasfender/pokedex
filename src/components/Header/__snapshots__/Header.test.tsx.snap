// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should initialize with default parameters 1`] = `
.c0 {
  padding: 1rem;
}

.c0 img {
  width: 17rem;
}

<header
  class="c0"
>
  <img
    alt=""
    src="/img/logo.png"
  />
</header>
`;
